doc/api/buffer.md

```diff
@@ -3277,6 +3277,42 @@ While, the `Buffer` object is available as a global, there are additional
 `Buffer`-related APIs that are available only via the `buffer` module
 accessed using `require('buffer')`.
 
+### `buffer.atob(data)`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `data` {any} The Base64-encoded input string.
+
+Decodes a string of Base64-encoded data into bytes, and encodes those bytes
+into a string using Latin-1 (ISO-8859-1).
+
+The `data` may be any JavaScript-value that can be coerced into a string.
+
+**This function is only provided for compatibility with web platform APIs
+and should only be used when necessary in code intended to be run on
+both the browser and Node.js. For code running only within Node.js,
+converting strings and {Buffer}s to and from Base64 using
+`Buffer.from(str, 'base64')` and `buf.toString('base64')` is faster.**
```

Comment:
The point isn’t performance or Web Compatibility – even if this were fast, people still shouldn’t use it, and it would still be considered a legacy API on the Web as well. (And we certainly should not be tempting ourselves to improve performance here.)

```suggestion
**This function is only provided for compatibility with legacy web platform APIs
and should never be used in new code.
For code running using Node.js APIs, converting between base64-encoded strings 
and binary data should be performed using `Buffer.from(str, 'base64')` and
`buf.toString('base64')`, as `Buffer`s (and, more generally, `Uint8Array`s)
are the proper language interfaces for working with binary data.**
```

Reply:
Yeah, I get that, but we should give a better reason why they shouldn't be used in new code, especially when there are currently no obvious web platform standard alternative APIs that are portable across environments. `Uint8Array` is great for working with binary data but does not offer Base64 conversion.

Reply:
I don’t think “the web doesn’t ship with a good API out of the box” is a good reason to recommend a bad API instead. There are plenty of libraries that solve this problem (including lots with `Buffer`-like interfaces), and we should be clear that this **is** a legacy API on all platforms which should not be used in new code and that that **is** because it’s doing the conceptually wrong thing.

Reply:
I don't disagree. I just would like to have more of an explanation why other than "don't do this"

Reply:
> I just would like to have more of an explanation why other than "don't do this"

@jasnell I’m not sure if you are asking for an explanation here, or making a statement – after all, you are suggesting to add these APIs here and I assume you’ve read up on their history beforehand – but that’s what the last sentence in my suggestion is about. I would not want to go too deep into details about the concept of “binary strings” in the Node.js docs, as it does feel somewhat tangential in the end, but the tl;dr is that strings are sequences of *characters*, not bytes, and that if you want to work with binary data, you’ll want sequences of bytes instead, and you’re bound to run into problems unless you know what you’re doing (in which case you wouldn’t be using `atob` or `btoa` to begin with).

It’s the same reason that we’ve renamed the `'binary'` character encoding and marked it as legacy. (There’s a link to link to https://developer.mozilla.org/en-US/docs/Web/API/DOMString/Binary there for that, and you can also add one here if you want.)

Reply:
I'm asking for a suggestion for more detail to add to the doc... as in "...and should never be used in new code *because ....*"

Reply:
How about “should never be used in new code, because they use strings to represent binary data, as they predate the introduction of typed arrays in JavaScript.”?

Reply:
Updated!

Reply:
The irony is that if you google something like "convert typed array to base64", it's difficult to find a result that doesn't suggest to convert the binary data to a binary string and use `btoa` on it :(